### PR TITLE
Submitting solution

### PR DESCRIPTION
**Pull request by cody-berry**
This is one of the worst solutions I've ever wrote. But it'll do.

### CODE REVIEW

File: 2697. Lexicographically Smallest Palindrome/solution.py [hunk 1]
```diff
@@ -5,21 +5,38 @@ class Solution:
     # the alphabet then the respective letter in b does.
     def makeSmallestPalindrome(self, s: str) -> str:
         # These are the two indices that, in the resulting string, we are going to make the same.
-        # If the string length is odd, then the first index should start at the floored value
-        # and the second index should start at the ceiled value.
+        # If the string length is odd, then the center should be the middle letter, and the first
+        # and last indices surround it.
+        if (len(s) % 2 == 1):
```
Comment (aeruru) [inline 2697. Lexicographically Smallest Palindrome/solution.py:10]: I think you probably could have simplified all of this logic with the integer divide operator (//) and negative array indices

File: 2697. Lexicographically Smallest Palindrome/solution.py [hunk 2]
```diff
@@ -5,21 +5,38 @@ class Solution:
...
+            if (s[firstIndex] != s[lastIndex]):
+                # If they are different, then check the code of the letters. Note that this is
                 # only lowercase English letters, so there will be no confusions.
+                if (s[firstIndex] > s[lastIndex]):
+                    smallestLetter = s[lastIndex]
+                else:
+                    smallestLetter = s[firstIndex]
                 
                 # Make the letter with the bigger code the one with the smaller code.
+                changedString = (changedString[:firstIndex] + smallestLetter +
```
Comment (aeruru) [inline 2697. Lexicographically Smallest Palindrome/solution.py]: This code is pretty bad and super inefficient as you create a new string with every modification. Do you remember the technique I used in my first solution to the first problem in this contest?

**Review by aeruru — CHANGES_REQUESTED** [review]
I think you can do better so requesting changes :)

File: 2697. Lexicographically Smallest Palindrome/solution.py [hunk 3]
```diff
@@ -5,21 +5,36 @@ class Solution:
...
-                # If they are different, then check the ascii code of the letters. Note that this is
+            if (s[firstIndex] != s[lastIndex]):
+                # If they are different, then check the code of the letters. Note that this is
                 # only lowercase English letters, so there will be no confusions.
+                if (s[firstIndex] > s[lastIndex]):
+                    smallestLetter = s[lastIndex]
+                else:
+                    smallestLetter = s[firstIndex]
                 
                 # Make the letter with the bigger code the one with the smaller code.
```
Comment (winry-fruitiwi) [inline 2697. Lexicographically Smallest Palindrome/solution.py:34]: What is "code"? Is it the ord?

File: 2697. Lexicographically Smallest Palindrome/solution.py [hunk 4]
```diff
@@ -5,21 +5,36 @@ class Solution:
...
+                changedString = f"{changedString[:firstIndex]} + smallestLetter + {changedString[(firstIndex+1):lastIndex]} + smallestLetter + {changedString[(lastIndex+1):]}"
```
Comment (winry-fruitiwi) [inline 2697. Lexicographically Smallest Palindrome/solution.py:35]: I can't understand this, unfortunately. I think you should explain this a bit better.

**Review by winry-fruitiwi — COMMENTED** [review]
Overall, I think your code is a bit messy and poorly documented.

**Pull request merged**